src/main/java/bob/Bob.java

```diff
@@ -0,0 +1,83 @@
+package bob;
+
+import java.io.IOException;
+
+import bob.commands.Command;
+import bob.common.Messages;
+import bob.data.task.Tasklist;
+import bob.exceptions.BobException;
+import bob.parser.Parser;
+import bob.storage.Storage;
+import bob.ui.Ui;
+
+
+/**
+ * Represents the task-managing ChatBot.
+ */
+public class Bob {
+    private final Storage storage;
+    private final Tasklist tasks;
+    private final Ui ui;
+
+    /**
+     * Creates a Bob.
+     *
+     * @throws IOException If saved file can't be loaded.
+     */
+    public Bob() throws IOException {
+        //File path of saved task list
+        String filePath = System.getProperty("user.dir").endsWith("text-ui-test")
+                ? "test.txt"
+                : System.getProperty("user.dir").endsWith("ip")
+                ? "data/bob.txt"
+                // Creates a save file on the user's home directory if user is not in ip directory
+                : System.getProperty("user.home") + "/bob.txt";
```

Comment:
This is cool but quite hard to read. Will if statements be more readable if you require more than 1 ternary operation?

Reply:
These statements are more for pathing of the file containing the saved txt. I can make the conditions into variables to improve on readability! Thanks for the suggestion. ^^